src/main/kotlin/net/maizegenetics/phgv2/utils/VariantLoadingUtils.kt

```diff
@@ -0,0 +1,280 @@
+        // use the -f option to overwrite any existing file
+        myLogger.info("bgzipping  file ${gvcfFileName}")
+        val gvcfGzippedFile = gvcfFileName + ".gz"
+        var builder = ProcessBuilder(
+            "bgzip", "-f", gvcfFileName)
+
+        var process = builder.start()
+        var error: Int = process.waitFor()
+        if (error != 0) {
+            myLogger.warning("\nERROR $error creating bgzipped  version of file: $gvcfFileName")
+            throw IllegalStateException("bgzipAndIndexGVCFfile: error trying to bgzip file ${gvcfFileName}: ${error}")
+        }
+
+        // File has been gzipped, now index it.
+        // Use the -f option to overwrite any existing index
+        // We will use bcftools to create the csi index
+        // ORiginal PHG used tabix, we wnat csi indexes to allow for large genomes e.g wheat.
+        // TileDB supports .csi indexed files.
+        val tabixFile = gvcfGzippedFile + ".csi"
+        builder = ProcessBuilder("bcftools", "index", "-c",gvcfGzippedFile)
+        process = builder.start()
+        error = process.waitFor()
+        if (error != 0) {
+            myLogger.warning("\nERROR $error creating tabix indexed  version of file: $gvcfGzippedFile")
+            throw IllegalStateException("bgzipAndIndexGVCFfile: error trying to run bcftools index -c on file ${gvcfGzippedFile}: ${error}")
+        }
+        return gvcfGzippedFile
+    } catch (exc:Exception) {
+        throw IllegalStateException("bgzipAndIndexGVCFfile: error bgzipping and/or indexing file ${gvcfFileName}")
+    }
+
+}
+
+/**
+ * Function to add a sequence Dictionary based on the reference genome.  This uses the loaded genome to get the lengths.
+ */
+fun addSequenceDictionary(vcfheader : VCFHeader, refGenomeSequence: Map<String,NucSeq>) {
+
+    val sequenceRecordList = refGenomeSequence.keys.map { SAMSequenceRecord(it,
+        refGenomeSequence[it]!!.size()) }
+
+    vcfheader.setSequenceDictionary(SAMSequenceDictionary(sequenceRecordList))
+}
+
+// Method to verify anchors for genome intervals have no overlapping positions
+// Overlapping intervals are not supported in the PHG
+fun verifyIntervalRanges(intervalFile: String): Set<String> {
+    val overlappingPositions: MutableSet<String> = HashSet() // overlaps to be returned
+    val intervalRanges: RangeSet<Position> = TreeRangeSet.create()
+    // Read the anchor file, store to RangeSet, check for overlaps as you add
+    // Store overlapping anchors to a Set to be returned to calling method
+    println("verifyIntervalRanges: checking file ${intervalFile} for overlaps")
+    try {
+        bufferedReader(intervalFile).use { br ->
+            var curLine = br.readLine()
+            while (curLine != null) {
+                if (curLine.uppercase(Locale.getDefault()).contains("CHROMSTART")) continue
+                val tokens =
+                    curLine.split("\\t".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
```

Comment:
Does this need to be a regex?  I think doing:
`curLine.split("\t").dropeLastWhile{}`

Is just as efficient as its a single char split.

Reply:
I changed it.